inc/user.class.php

```diff
@@ -3404,173 +3402,243 @@ static function getSqlSearchResult ($count = true, $right = "all", $entity_restr
+                     ];
                }
                if (in_array($r, Profile::$helpdesk_rights)) {
                   $forcecentral = false;
                }
             }
 
-            $where = '('.implode(' OR ', $where);
-
             if ($forcecentral) {
-               $where .= " AND `glpi_profiles`.`interface` = 'central' ";
+               $WHERE['glpi_profiles.interface'] = 'central';
             }
-            $where .= ')';
       }
 
       if (!$inactive_deleted) {
-         $where .= " AND `glpi_users`.`is_deleted` = 0
-                     AND `glpi_users`.`is_active` = 1
-                     AND (`glpi_users`.`begin_date` IS NULL
-                          OR `glpi_users`.`begin_date` < NOW())
-                     AND (`glpi_users`.`end_date` IS NULL
-                          OR `glpi_users`.`end_date` > NOW())";
+         $WHERE = array_merge(
+            $WHERE, [
+               'glpi_users.is_deleted' => 0,
+               'glpi_users.is_active'  => 1,
+               [
+                  'OR' => [
+                     ['glpi_users.begin_date' => null],
+                     ['glpi_users.begin_date' => ['<', new QueryExpression('NOW()')]]
+                  ]
+               ],
+               [
+                  'OR' => [
+                     ['glpi_users.end_date' => null],
+                     ['glpi_users.end_date' => ['>', new QueryExpression('NOW()')]]
+                  ]
+               ]
+
+            ]
+         );
       }
 
       if ((is_numeric($value) && $value)
           || count($used)) {
 
-         $where .= " AND `glpi_users`.`id` NOT IN (";
-         if (is_numeric($value)) {
-            $first  = false;
-            $where .= $value;
-         } else {
-            $first = true;
-         }
-         foreach ($used as $val) {
-            if ($first) {
-               $first = false;
-            } else {
-               $where .= ",";
-            }
-            $where .= $val;
-         }
-         $where .= ")";
+         $WHERE[] = [
+            'NOT' => [
+               'glpi_users.id' => $used
+            ]
+         ];
       }
 
+      $criteria = [
+         'SELECT'    => 'glpi_users.*',
+         'FROM'      => 'glpi_users',
+         'LEFT JOIN' => [
+            'glpi_useremails'       => [
+               'ON' => [
+                  'glpi_useremails' => 'users_id',
+                  'glpi_users'      => 'id'
+               ]
+            ],
+            'glpi_profiles_users'   => [
+               'ON' => [
+                  'glpi_profiles_users'   => 'users_id',
+                  'glpi_users'            => 'id'
+               ]
+            ]
+         ]
+      ];
       if ($count) {
-         $query = "SELECT COUNT(DISTINCT `glpi_users`.`id` ) AS CPT
-                   FROM `glpi_users` ";
-      } else {
-         $query = "SELECT DISTINCT `glpi_users`.*
-                   FROM `glpi_users` ";
+         $criteria['COUNT'] = 'CPT';
```

Comment:
Same remark here : https://www.db-fiddle.com/f/7Q39KHLyxMbh7KsYsypar2/1

Reply:
Yes, this is the same as the previous now ;)